agent/accumulator.go

```diff
@@ -161,6 +164,14 @@ func (ac *accumulator) AddFields(
 	ac.metrics <- m
 }
 
+func (ac *accumulator) AddError(err error) {
+	if err == nil {
+		return
+	}
+	atomic.AddUint64(&ac.errCount, 1)
+	log.Printf("ERROR in input [%s]: %s", ac.inputConfig.Name, err)
```

Comment:
The only place where `inputConfig.Name` is touched is in [internal/config/config.go:798](https://github.com/influxdata/telegraf/blob/4363eebc1b2da873a76f770abcf5e9aad2d1879a/internal/config/config.go#L798) (during initialization). It is never read or written to ever again (prior to this PR it was an unused variable). Thus there's no need to protect it.